Sanatize the creation of temp files in tests

DunderRoffe opened the pull request:
Signed-off-by: Viktor Sjölind <viktor.sjolind@pelagicore.com>

Commit:
Sanatize the creation of temp files in tests

Signed-off-by: Viktor Sjölind <viktor.sjolind@pelagicore.com>

kursatkobya commented on libsoftwarecontainer/component-test/softwarecontainer_test.cpp:
```diff
@@ -78,29 +78,70 @@ void SoftwareContainerTest::TearDown()
+    std::string strWithPrefix(prefix + "/tmp/SC-tmpFileXXXXXX");
+    size_t n = strWithPrefix.size() + 1;
+    char *fileTemplate = new char[n];
+    strncpy(fileTemplate, strWithPrefix.c_str(), n);
```
I think this 4 line is repeated down as well maybe it worth to create a function for it.

kursatkobya commented on libsoftwarecontainer/component-test/softwarecontainer_test.cpp:
```diff
@@ -78,29 +78,70 @@ void SoftwareContainerTest::TearDown()
+    std::string strWithPrefix(prefix + "/tmp/SC-tmpDirXXXXXX");
+    size_t n = strWithPrefix.size() + 1;
+    char *dirTemplate = new char[n];
+    strncpy(dirTemplate, strWithPrefix.c_str(), n);
```
also copy function can be used here 
`strWithPrefix.copy(dirTemplate, strWithPrefix.size())`

kursatkobya commented on libsoftwarecontainer/component-test/softwarecontainer_test.cpp:
```diff
@@ -78,29 +78,70 @@ void SoftwareContainerTest::TearDown()
 {
-    char fileTemplate[] = "/tmp/SC-tmpXXXXXX";
-    int fd = 0;
+    std::string name;
```
it could be nice if you can initialize it here, maybe using a lambda like ...
```
std::string name = [&]{
    if (directory) {
        return createTempDir();
   } else {
        return createTempFile();
   }
} ();
```

tobsan replied:
But you would still need the checks later to do unlink/rmdir, or you would do them in the lambda

Commit:
Fixes tempfile/dir creation + unix socket test

mkstemp/mkdtemp will fail if one tries to create a temporary file/dir
in a sub-directory that does not yet exist. We therefore only prepend
with /tmp if there is no prefix already in those functions.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>
